GAE/src/org/waterforpeople/mapping/app/web/DeviceNotificationRestServlet.java

```diff
@@ -92,12 +98,42 @@ protected void writeOkResponse(RestResponse resp) throws Exception {
             missingUnknown.put(mu);
         }
 
+        for (Long id : r.getDeletedSurveys()) {
+            deletedSurveys.put(String.valueOf(id));
+        }
+
         json.put("missingFiles", missingFiles);
         json.put("missingUnknown", missingUnknown);
+        json.put("deletedForms", deletedSurveys);
 
         getResponse().getWriter().println(json.toString());
     }
 
+    private Set<Long> getDeletedSurveys(DeviceNotificationRequest req) {
+        Set<Long> surveyIds = new HashSet<>();
+        String[] args = req.getSurveyIds();
```

Comment:
My understanding is that is responsibility of `populateFields` to do this conversion of `String[]` to `List<Long>`